requirements.txt

```diff
@@ -1,14 +1,14 @@
 # Runtime requirements for the python 3 version of cirq.
 
 dataclasses; python_version < '3.7'
-google-api-python-client~=1.6
-matplotlib~=3.0
-networkx==2.3
-numpy~=1.16
+google-api-python-client~=1.7
+matplotlib~=3.1
+networkx==2.4
+numpy~=1.17
```

Comment:
Don't bump user version numbers in requirements.txt unless we actually *require* that version for the code to work. There should be some bug that was fixed or some feature we need. Otherwise we can create incompatibilities with other libraries e.g. if they require a specific version of numpy because the latest one is broken.

Reply:
incompatibilities were there before that prevented asteroids dependency from installing. 
the incompatibilities were coming from hard pinned specific versions. 
i have not documented that. 
i think i will split the pr in two branches, one for dependencies and one for pylint format fixes,